src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,223 @@
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT c.id as car_id, c.model, m.name as manufacturer_name, "
+                + "m.id as manufacturer_id, m.country as manufacturer_country"
+                + " FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.id = ? AND c.is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement = connection.prepareStatement(query)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCarWithManufacturer(resultSet);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't get car from DB by id " + id + " ",
+                    throwable);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversList(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id as car_id, c.model  as car_model, m.name as manufacturer_name, "
+                + "m.id as manufacturer_id,"
+                + " country as manufacturer_country FROM cars c JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getAllCars = connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllCars.executeQuery();
+            while (resultSet.next()) {
+                Car car = getCarWithManufacturer(resultSet);
+                cars.add(car);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't get all cars from DB by id ",
+                    throwable);
+        }
+        driversToListOfCars(cars);
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in cars DB.", throwable);
+        }
+        removeDrivers(car);
+        List<Driver> drivers = car.getDrivers();
+        if (drivers != null) {
```

Comment:
this check is redundant, if app is correct you shouldn't have car with null drivers

Reply:
done